### PR TITLE
Made Classes public and overridable for custom implementations

### PR DESCRIPTION
**Pull request by Patrick-Spiegel**
This pullrequest publishes classes required to wrote a custom FilterExpressionCompiler.

### CODE REVIEW

File: net/DevExtreme.AspNet.Data/DataSourceLoaderImpl.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public DataSourceLoaderImpl(IQueryable<S> source, DataSourceLoadOptionsBase opti
 #endif
         }
 
-
```
[the remark this answers is not in the document]
Comment (AlekseyMartynov) [reply net/DevExtreme.AspNet.Data/DataSourceLoaderImpl.cs]: It is now fixed. However, there are failing unit tests: https://ci.appveyor.com/project/dxrobot/devextreme-aspnet-data/builds/30187012/tests

**Comment by AlekseyMartynov** [comment]
Thanks for your input. Would you please share your use case for custom filter compilers? We had no plans to publish implementation details, such as `Utils` or `AnonTypeNewTweaks`. Let me see if I can suggest an alternative solution.

**Comment by Patrick-Spiegel** [comment]
https://www.devexpress.com/Support/Center/Question/Details/T851638/how-to-use-a-custom-filter-on-the-devextreme-aspnet-library

I'd like to use a custom filter for querying data with RavenDB. RavenDB does not support String.contains and needs to be used with IRavenQueryable.Search.

**Comment by AlekseyMartynov** [comment]
May I ask you to share your custom filter compiler implementation that solves the `contains` issue discussed in T851638?

The main difficulty is that Data Grid blends the search value into the composite filter. It is not trivial to extract it back in the general case. Therefore, the suggested approach with an external text box and a custom parameter added in [`onBeforeSend`](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#configuration) looks easier and more robust. Do you need a code sample to illustrate this apprach?

**Comment by Patrick-Spiegel** [comment]
I have made the ticket private and posted a new answer there.

**Comment by AlekseyMartynov** [comment]
Thank you. We will review it shortly.

**Pull request closed without merging**